adaptive/src/main/java/com/google/accompanist/adaptive/TwoPane.kt

```diff
@@ -0,0 +1,598 @@
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.graphics.toComposeRect
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.layout.LayoutCoordinates
+import androidx.compose.ui.layout.boundsInWindow
+import androidx.compose.ui.layout.layoutId
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.constrain
+import androidx.compose.ui.unit.constrainHeight
+import androidx.compose.ui.unit.constrainWidth
+import androidx.compose.ui.unit.dp
+import androidx.window.layout.DisplayFeature
+import androidx.window.layout.FoldingFeature
+import kotlin.math.roundToInt
+
+/**
+ * A layout that places two different pieces of content defined by the [first] and [second]
+ * slots where the arrangement, sizes and separation behaviour is controlled by [TwoPaneStrategy].
+ *
+ * [TwoPane] is fold and hinges aware using the provided the [displayFeatures] (which should
+ * normally be calculated via [calculateDisplayFeatures]). The layout will be adapted to properly
+ * separate [first] and [second] panes so they don't interfere with hardware hinges (vertical or
+ * horizontal) as specified in [displayFeatures], or respect folds when needed (for example, when
+ * foldable is half-folded (90-degree fold AKA tabletop) the split will become on the bend).
+ *
+ * To only be aware of folds with a specific orientation, pass in an alternate
+ * [foldAwareConfiguration] to only adjust for vertical or horizontal folds.
+ *
+ * The [TwoPane] layout will always place both [first] and [second], based on the provided
+ * [strategy] and window environment. If you instead only want to place one or the other,
+ * that should be controlled at a higher level and not calling [TwoPane] if placing both is not
+ * desired.
+ *
+ * @param first the first content of the layout, a left-most in LTR, a right-most in RTL and
+ * top-most in a vertical split based on the [SplitResult] of [TwoPaneStrategy.calculateSplitResult]
+ * @param second the second content of the layout, a right-most in the LTR, a left-most in the RTL
+ * and the bottom-most in a vertical split based on the [SplitResult] of
+ * [TwoPaneStrategy.calculateSplitResult]
+ * @param strategy strategy of the two pane that controls the arrangement of the layout
+ * @param modifier an optional modifier for the layout
+ */
+@Composable
+public fun TwoPane(
+    first: @Composable () -> Unit,
+    second: @Composable () -> Unit,
+    strategy: TwoPaneStrategy,
+    displayFeatures: List<DisplayFeature>,
+    modifier: Modifier = Modifier,
+    foldAwareConfiguration: FoldAwareConfiguration = FoldAwareConfiguration.AllFolds,
```

Comment:
Curious why this is not on the public TwoPaneStrategies? Do you think it would be a better API?

Reply:
If this was on each of the `TwoPaneStrategy` s, it would be a bit more awkward to define a custom `TwoPaneStrategy` since it would have to be fold aware. Placing it on the single `TwoPane` entry point means that at least one type of fold has to be handled, and the rest of the strategies don't need to directly care about folds.